environment/environment-duration/duration-policies.rego

```diff
@@ -2,20 +2,223 @@ package torque.environment
+}
 
-result := { "decision": "Denied", "reason": "sandbox must have duration" } if {
-    not contains(input, "duration")
+get_timespan_string(max_duration_ns) = timespan {
+    minutes_str = minutes(max_duration_ns)
+	not days(max_duration_ns)
+    not hours(max_duration_ns)
+	timespan := concat(" ", [minutes_str])
 }
 
-result = { "decision": "Denied", "reason": "sandbox duration exceeds max duration" } if {
-    data.max_duration_minutes < input.duration_minutes
+get_timespan_string(max_duration_ns) = timespan {
+	days_str = days(max_duration_ns)
+	minutes_str = minutes(max_duration_ns)
+    not hours(max_duration_ns)
+	timespan := concat(" ", [days_str, minutes_str])
 }
 
-result = { "decision": "Manual", "reason": "sandbox duration requires approval" } if {
+get_timespan_string(max_duration_ns) = timespan {
+	days_str = days(max_duration_ns)
+	hours_str = hours(max_duration_ns)
+    not minutes(max_duration_ns)
+	timespan := concat(" ", [days_str, hours_str])
+}
+
+get_timespan_string(max_duration_ns) = timespan {
+	hours_str = hours(max_duration_ns)
+	minutes_str = minutes(max_duration_ns)
+    not days(max_duration_ns)
+	timespan := concat(" ", [hours_str, minutes_str])
+}
+
+get_timespan_string(max_duration_ns) = timespan {
+	days_str = days(max_duration_ns)
+    not hours(max_duration_ns)
+    not minutes(max_duration_ns)
+	timespan := concat(" ", [days_str])
+}
+
+get_timespan_string(max_duration_ns) = timespan {
+	hours_str = hours(max_duration_ns)
+    not days(max_duration_ns)
+    not minutes(max_duration_ns)
+	timespan := concat(" ", [hours_str])
+}
+
+get_timespan_string(max_duration_ns) = timespan {
+	minutes_str = minutes(max_duration_ns)
+    not days(max_duration_ns)
+    not hours(max_duration_ns)
+	timespan := concat(" ", [minutes_str])
+}
+
+
+# This policy enforces a maximal duration on environments launched from it.
+# It takes two numbers as arguments (in the data object):
+#   1. env_max_duration_minutes : the total number of minutes the environments are allowed to run (including extentions)
+#   2. env_duration_for_manual_approval_minutes : the duration of the environment which will require a manual approval.
+# Usually, env_duration_for_manual_approval_minutes will be smaller than max_duration_minutes.
+#
+# An example of a data object for this policy looks like this:
+# {
+#   "max_duration_minutes": 300,
+#   "duration_for_manual_minutes": 120
+# }
+#
+# An example with the new variable names
+# {
+#   "env_max_duration_minutes": 300,
+#   "env_duration_for_manual_approval_minutes": 120
+# }
+#
+# In this example we allow the environment to be up to 5 hours total. Anything more than 5 hours would be automatically denied.
+# Duration which is between 2 and 5 hours would need manual approval,
+# and below 2 hours would be automatically approved without manual intervention.
+result := { "decision": "Denied", "reason": "environment must have duration" } if {
+    not input.duration_minutes
```

Comment:
why do we need validations on the inputs? Torque passes it, no need to verify it
even validations on the data can be argued. maybe we can do it in the code. will be easier and also will be static validations and will not fail only on runtime